### DISCUSSION TITLE
Stable filenames

### DISCUSSION THREAD

**Discussion by lsli8888** [rbuchberger/jekyll_picture_tag#268] [Q&A]
I may have missed something in the documentation, but is there a way of generating images with predictable filenames without the nine random characters (hash?) that appears right before the file extension?

For example, let's say I have an image named mypic.jpg and I want to generate images 200px and 400px wide.  The plugin will currently generate something like:

mypic-200-4480c9eda.jpg
mypic-400-4480c9eda.jpg

When instead, I want something like:

mypic-200.jpg
mypic-400.jpg

Is this possible?

**Comment by rbuchberger** [comment]
Currently it's not possible. The reason files are named that way is because that's how we know if either the file or any of the settings used to generate it have changed. Those random characters are a shortened hash of the source image and all the relevant settings. This way, if any of them changes, the filename changes and it will get regenerated.

It might be possible to store this information in a file somewhere instead, but that would be a moderately difficult change to implement.
